core/src/modules/configLoaders/app.ts

```diff
@@ -17,16 +19,21 @@ export async function loadApp(
   let isNew = false;
   validateConfigObjectKeys(App, configObject);
 
+  // We assume we will always have to create a new object when in config mode,
+  // so it is safe to leave locked in the find query.
```

Comment:
Dropping these in every loader. Happy to remove if you feel like it is unnecessary. The locked logic was starting to feel a little tangled, so I did this to help future devs.

Reply:
I'm not sure these comments help me. maybe something in the `ConfigWriter.getLockKey` function instead as to why it is null.